platform-ajax/src/main/java/com/softicar/platform/ajax/engine/AjaxDomEngine.java

```diff
@@ -427,22 +421,21 @@ public void executeScriptCode(String scriptCode) {
 	// -------------------------------- forms -------------------------------- //
 
 	@Override
-	public void resetForm(IDomNode form) {
+	public void triggerUploadOnChange(IDomNode form, IDomNode triggerNode) {
 
-		JS_callNodeFunction(form, "reset");
+		triggerUploadOnEvent(form, triggerNode, "onchange");
 	}
 
 	@Override
-	public void submitForm(IDomNode form) {
+	public void triggerUploadOnClick(IDomNode form, IDomNode triggerNode) {
 
-		JS_call("c.submitForm", form.getNodeId());
+		triggerUploadOnEvent(form, triggerNode, "onclick");
 	}
 
-	@Override
-	public void submitFormOnChange(IDomNode form, IDomNode triggerNode) {
+	private void triggerUploadOnEvent(IDomNode form, IDomNode triggerNode, String event) {
```

Comment:
- I removed the method `resetForm`  since I see no point providing it.
- I removed `submitForm` because it was not used, is not necessary, and does not really do what is advertised. It actually executed an AJAX upload request.
- I renamed `submitFormOnChange` to ` triggerUploadOnEvent` since it the original name was a misnomer.